### PR TITLE
ergo: update to 2.11.1, rename from oragono.

### PR DESCRIPTION
**Pull request by chilledfrogs**
**Note**: I couldn't find out what was the procedure for updating packages where they get renamed, I did what I thought wasn't stupid, please feel free to tell me if I screwed anything up with it :sweat_smile:

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64

### CODE REVIEW

**Comment by TinfoilSubmarine** [comment]
I think you need a dummy package, see e.g. `mysql` which is symlinked to `mariadb` and has the `provides=...` and `replaces=...` for mysql.

**Comment by TinfoilSubmarine** [comment]
Sorry, commented before I was done thinking :)

You already have the symlink, but I think the `provides=...` and `replaces=...` should be similar to what `mysql`/`mariadb` has since this is just a rename, as well as an `oragano_package()` similar to `mysql_package()`.

**Comment by TinfoilSubmarine** [comment]
Actually, you should follow what was done for `pulseeffects` being renamed to `easyeffects`, since that was more recent and was just a rename unlike the `mysql`/`mariadb` drop-in case.

File: srcpkgs/ergo/files/ergo/log/run [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec vlogger -t ergo
```
Comment (ericonr) [inline srcpkgs/ergo/files/ergo/log/run:2]: Please use the symlink approach instead.

File: srcpkgs/ergo/INSTALL.msg [hunk 2]
```diff
@@ -0,0 +1 @@
+Oragono was renamed to Ergo since version 2.7.0.
```
Comment (ericonr) [inline srcpkgs/ergo/INSTALL.msg:1]: It's missing a trailing newline

And I think it should be `oragono.INSTALL.msg` so only the people using the old package get the warning.

**Review by ericonr — COMMENTED** [review]
@pbui ok for you?

**Comment by pbui** [comment]
> @pbui ok for you?

Yes, these changes look reasonable.

As a side note, you can probably remove me from the maintainer field... as I don't use Void Linux anymore and probably will not be contributing updates for the foreseeable future.

File: srcpkgs/ergo/INSTALL.msg (hunk 2 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/INSTALL.msg:1]: I see, will do in just a minute

File: srcpkgs/ergo/files/ergo/log/run (hunk 1 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/files/ergo/log/run:2]: Whoops, true 😅

File: srcpkgs/ergo/template [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Template file for 'ergo'
+pkgname=ergo
+version=2.9.1
+revision=1
+build_style=go
+go_import_path="github.com/ergochat/ergo"
+hostmakedepends="git"
```
Comment (paper42) [inline srcpkgs/ergo/template:7]: git is most likely not necessary

File: srcpkgs/ergo/template [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+short_desc="Modern IRC server (daemon/ircd) written in Go"
+maintainer="Allen Sobot <chilledfrogs@disroot.org>"
+license="MIT"
+homepage="https://ergo.chat/"
```
Comment (paper42) [inline srcpkgs/ergo/template:11]:
```suggestion
homepage="https://ergo.chat/"
changelog="https://raw.githubusercontent.com/ergochat/ergo/master/CHANGELOG.md"
```

File: srcpkgs/ergo/template (hunk 3 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/template:7]: Will try without, I thought I saw it written to always add that for Go stuff

File: srcpkgs/ergo/template (hunk 4 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/template:11]: Done

File: srcpkgs/ergo/template (hunk 3 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/template]: Worked without

File: srcpkgs/ergo/template [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Template file for 'ergo'
+pkgname=ergo
+version=2.9.1
+revision=1
+build_style=go
+go_import_path="github.com/ergochat/ergo"
+#hostmakedepends="git"
```
Comment (paper42) [inline srcpkgs/ergo/template:7]: commenting this out doesn't make much sense, let's remove it

Comment (chilledfrogs) [reply srcpkgs/ergo/template:7]: Whoops, terribly sorry, wanted to do that but forgot

**Comment by sgn** [comment]
Please move data files in `/var` to new location if we choose to rename its folder.

**Comment by sgn** [comment]
> @sgn I'm sorry but I'm not sure what you want me to do concretely with which exact folder

I meant move files from `/var/lib/oragono` to `/var/lib/ergo`

**Comment by chilledfrogs** [comment]
@sgn I actually wanted to do that conditionally but couldn't figure out whether there was an appropriate way to do it or not, so any ideas would be greatly appreciated

**Comment by paper42** [comment]
> @sgn I actually wanted to do that conditionally but couldn't figure out whether there was an appropriate way to do it or not, so any ideas would be greatly appreciated

What would be the condition? Do you mean that you would want existing installations to keep using the old location, but new ones to use the ergo location? We don't have a good way to do this.

**Comment by sgn** [comment]
In my opinion, at least keychain should be kept.

**Comment by chilledfrogs** [comment]
@paper42 I more meant as a specific way of doing a one-time rename of the directory when updating it

**Comment by chilledfrogs** [comment]
Would something like the `oragono.INSTALL` I made be appropriate?

**Comment by Piraty** [comment]
applied onto 3a4b215f05ee04652cb5102b85b7d4c08d7aa160, it currently FBTFS .
```
_cgo_callers: relocation target x_cgo_callers not defined
_cgo_init: relocation target x_cgo_init not defined
_cgo_mmap: relocation target x_cgo_mmap not defined
_cgo_munmap: relocation target x_cgo_munmap not defined
/usr/lib/go/pkg/tool/linux_amd64/link: too many errors
=> ERROR: ergo-2.10.0_1: do_build: 'go install -p "$XBPS_MAKEJOBS" -mod="${go_mod_mode}" -x -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}' exited with 2
=> ERROR:   in do_build() at common/build-style/go.sh:43
```

**Comment by chilledfrogs** [comment]
> applied onto [3a4b215](https://github.com/void-linux/void-packages/commit/3a4b215f05ee04652cb5102b85b7d4c08d7aa160), it currently FBTFS .
Terribly sorry but FBTFS? :sweat_smile:

> ```
> _cgo_callers: relocation target x_cgo_callers not defined
> _cgo_init: relocation target x_cgo_init not defined
> _cgo_mmap: relocation target x_cgo_mmap not defined
> _cgo_munmap: relocation target x_cgo_munmap not defined
> /usr/lib/go/pkg/tool/linux_amd64/link: too many errors
> => ERROR: ergo-2.10.0_1: do_build: 'go install -p "$XBPS_MAKEJOBS" -mod="${go_mod_mode}" -x -tags "${go_build_tags}" -ldflags "${go_ldflags}" ${go_package}' exited with 2
> => ERROR:   in do_build() at common/build-style/go.sh:43
Issue is I can't reproduce it for the moment and *seems* apparently neither can the CI, but is it related to this? https://github.com/golang/go/issues/43505

File: srcpkgs/ergo/oragono.INSTALL [hunk 6]
```diff
@@ -0,0 +1,10 @@
+case "$ACTION" in
+post)
+	if [ -f etc/oragono.conf ]; then
```
Comment (Piraty) [inline srcpkgs/ergo/oragono.INSTALL:3]: please test `$UPDATE = "yes"` here too

File: srcpkgs/ergo/files/ergo/log/run [hunk 7]
```diff
@@ -0,0 +1 @@
+/usr/bin/vlogger
```
Comment (Piraty) [inline srcpkgs/ergo/files/ergo/log/run:1]: please `exec`, like all other `log/run`

**Comment by chilledfrogs** [comment]
Wait, since `oragono` suddenly disappeared, what should I do? Bring it back as symlinks? @Piraty

File: srcpkgs/ergo/files/ergo/log/run (hunk 7 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/files/ergo/log/run:1]: It's just a symlink to `vlogger`, as I saw some other packages do, is this undesirable now?

File: srcpkgs/ergo/oragono.INSTALL (hunk 6 above)
Comment (chilledfrogs) [reply srcpkgs/ergo/oragono.INSTALL:3]: Like so (new commit)? :sweat_smile:

File: srcpkgs/ergo/files/ergo/log/run (hunk 7 above)
Comment (Piraty) [reply srcpkgs/ergo/files/ergo/log/run:1]: sorry i wasn't aware. also see https://github.com/void-linux/void-packages/pull/42026

Comment (chilledfrogs) [reply srcpkgs/ergo/files/ergo/log/run:1]: Fair enough, but concretely what do I go for then? Symlink or separate `vlogger` call?

Comment (classabbyamp) [reply srcpkgs/ergo/files/ergo/log/run:1]: rebase and you shouldn't need to do either :)

Comment (chilledfrogs) [reply srcpkgs/ergo/files/ergo/log/run:1]: I noticed something was done about more consistent logging, will do asap

**Comment by chilledfrogs** [comment]
...?

**Comment by chilledfrogs** [comment]
@classabbyamp could this be re-opened? To the best of my knowledge this PR still works just fine and is simply waiting to be merged for a very long time

**Comment by classabbyamp** [comment]
apparently not... make a new pr please an link this one. I've been meaning to look at this but it's fairly complex so hard to find the time

**Pull request closed without merging**